src/main/java/org/opensearch/knn/index/codec/nativeindex/IndexBuildHelper.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.nativeindex;
+
+import lombok.experimental.UtilityClass;
+import org.opensearch.knn.index.codec.nativeindex.model.BuildIndexParams;
+import org.opensearch.knn.index.quantizationService.QuantizationService;
+import org.opensearch.knn.index.vectorvalues.KNNVectorValues;
+import org.opensearch.knn.quantization.models.quantizationOutput.QuantizationOutput;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+
+import java.io.IOException;
+
+@UtilityClass
+class IndexBuildHelper {
+
+    /**
+     * Processes and returns the vector based on whether quantization is applied or not.
+     *
+     * @param knnVectorValues the KNN vector values to be processed.
+     * @param indexBuildSetup the setup containing quantization state and output, along with other parameters.
+     * @return the processed vector, either quantized or original.
+     * @throws IOException if an I/O error occurs during processing.
+     */
+    static Object processAndReturnVector(KNNVectorValues<?> knnVectorValues, IndexBuildSetup indexBuildSetup) throws IOException {
```

Comment:
why not do this?
```suggestion
    static <T> T processAndReturnVector(KNNVectorValues<T> knnVectorValues, IndexBuildSetup indexBuildSetup) throws IOException {
```

Reply:
I can't make it T , If I will make this T and the method also has to be T and Class TOO , not only here , Native Index Build strategy and Default also has to be Type erasure and then it will go to NativeindexWrite ...and I did not want to make ? so I would keep as it is.